torchtitan/metrics.py

```diff
@@ -11,13 +11,14 @@
 
 import torch
 from torch.utils.tensorboard import SummaryWriter
+from torchtitan import DEVICE_TYPE, DEVICE_MODULE
 from torchtitan.config_manager import JobConfig
 from torchtitan.logging import logger
 from torchtitan.parallelisms import ParallelDims
 
 # named tuple for passing GPU memory stats for logging
-GPUMemStats = namedtuple(
-    "GPUMemStats",
+MemStats = namedtuple(
```

Comment:
for consistency, let's name it to `DeviceMemStats`, and also similar for other occurrences of `mem_monitor`, `build_memory_monitor`